sdks/python/apache_beam/runners/dataflow/dataflow_job_service.py

```diff
@@ -34,31 +32,22 @@ def _invoke_runner(self):
     """Actually calls Dataflow and waits for completion.
     """
     runner = dataflow_runner.DataflowRunner()
-    result = runner.run_pipeline(
+    self.result = runner.run_pipeline(
         None, self.pipeline_options(), self._pipeline_proto)
     # Prefer this to result.wait_until_finish() to get state updates
     # and avoid creating an extra thread (which also messes with logging).
     dataflow_runner.DataflowRunner.poll_for_job_completion(
         runner,
-        result,
+        self.result,
         None,
         lambda dataflow_state: self.set_state(
             portable_runner.PipelineResult.pipeline_state_to_runner_api_state(
-                result.api_jobstate_to_pipeline_state(dataflow_state))))
-    return result
+                self.result.api_jobstate_to_pipeline_state(dataflow_state))))
+    return self.result
 
-  def pipeline_options(self):
-    def from_urn(key):
-      assert key.startswith('beam:option:')
-      assert key.endswith(':v1')
-      return key[12:-3]
-
-    return pipeline_options.PipelineOptions(
-        **{
-            from_urn(key): value
-            for (key, value
-                 ) in job_utils.struct_to_dict(self._pipeline_options).items()
-        })
+  def cancel(self):
+    if not self.is_terminal_state(self.state):
+      self.result.cancel()
```

Comment:
Are we using this function anywhere/is there a reason to pull this out?

Reply:
This overrides/implements the default no-op version at https://github.com/apache/beam/blob/v2.39.0/sdks/python/apache_beam/runners/portability/local_job_service.py#L313